code/datums/components/submerge_mod.dm

```diff
@@ -0,0 +1,55 @@
+//Allows an atom to contribute to the submerging effects of a turf
+//Submerge height is additive, so best to avoid spamming multiple atoms on a turf with this component/very high values unless you want wack results.
+/datum/component/submerge_modifier
+	///Height in pixels a target is submerged by the parent of this component
+	var/submerge_height
+
+/datum/component/submerge_modifier/Initialize(_submerge_height = 10)
+	if(!isatom(parent) || isturf(parent))
```

Comment:
Do you want this to work on areas? If it's just for object/mob you could check A/M instead

Reply:
There was a reason for this, but the code has changed since then so yeah I think this is redundant now.